### PR TITLE
Allow non-root user in config.json to work

### PR DESCRIPTION
**Pull request by cawallin**
When we updated presto-admin to make the config files owned by
presto:presto, because they have passwords, we neglected
to use sudo to access the files (which is necessary if the
user in config.json is not either root or the presto user)

### CODE REVIEW

**Comment by cawallin** [comment]
This still needs automated tests -- but it was tested manually. It also must be tested on EMR before it is merged.

File: prestoadmin/collect.py [hunk 1]
```diff
@@ -209,10 +210,10 @@ def system_info():
 
 def get_system_info(download_location):
 
-    if not exists(TMP_PRESTO_DEBUG):
-        run("mkdir " + TMP_PRESTO_DEBUG)
+    if not exists(TMP_PRESTO_DEBUG_REMOTE):
```
Comment (akshatnair) [inline prestoadmin/collect.py]: this seems unrelated

Comment (cawallin) [inline prestoadmin/collect.py]: it actually is related -- on the master, /tmp/presto-debug has already been created and is owned by root. when presto-admin logs into all the nodes and tries to create /tmp/presto-debug/sysinfo.txt with a non-root user, it fails

**Comment by cawallin** [comment]
Tests added

File: prestoadmin/collect.py (hunk 1 above)
Comment (akshatnair) [inline prestoadmin/collect.py]: but why was it already created as root? May be leftover from some other code?  Shouldn't all these be created with the same user as this is part of collect logs collect system_info

Comment (cawallin) [inline prestoadmin/collect.py]: it's created by the user that's running presto-admin, which right now always is root since you have to run presto-admin as sudo.

collect system_info is the only of the "collect" commands that needs to write out a file when executing a command on the remote nodes, and that remote command is run as the username you specify for sudo (which may not be root)

File: tests/unit/test_server.py [hunk 2]
```diff
@@ -323,14 +323,14 @@ def test_collect_node_information(self, mock_ext_ip, mock_service,
         self.assertEqual(('IP4', False, ''),
                          server.collect_node_information())
 
-    @patch('prestoadmin.server.run')
+    @patch('prestoadmin.server.sudo')
     def test_get_external_ip(self, mock_nodeuuid):
```
Comment (akshatnair) [inline tests/unit/test_server.py]: why is this called mock_uuid? and not mock_run and now mock_sudo?

**Comment by akshatnair** [comment]
lgtm mod small comments.

File: tests/unit/test_server.py (hunk 2 above)
Comment (cawallin) [inline tests/unit/test_server.py]: it's the command to get the node uuid, seems fine to keep it as is

**Comment by ebd2** [comment]
Did you look at the use of `put` in script.py and the several places that use `put_secure`?

Apart from that question, this looks good to me.

**Comment by cawallin** [comment]
I was assuming that you'd put the script in a place that the user has access to, it's run via sudo. I covered all of the places that use put_secure.

Works fine on EMR, I'm going to merge this.
